### PR TITLE
bridge: T6675: VXLAN Interface configuration lost due to improper bridge detachment

### PR DESCRIPTION
**Pull request by natali-rs1985**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6675
## Component(s) name

interfaces bridge
## How to test

```
set interfaces bridge br0 member interface vxlan4000
set interfaces vxlan vxlan4000 parameters neighbor-suppress
set interfaces vxlan vxlan4000 source-address '192.168.134.186'
set interfaces vxlan vxlan4000 vni '4000'
commit

vyos@vyos# del interfaces bridge br0 member interface vxlan4000
[edit]
vyos@vyos# commit
[ interfaces bridge br0 ]
To detach interface vxlan4000 from bridge you must first disable
"neighbor-suppress" parameter in the VXLAN interface vxlan4000

[[interfaces bridge br0]] failed
Commit failed
[edit]
vyos@vyos# del interfaces vxlan vxlan4000 parameters neighbor-suppress
[edit]
vyos@vyos# commit
[edit]
vyos@vyos#
```
## Smoketest result

```
vyos@vyos:~$ TEST_ETH="eth1 eth3" /usr/libexec/vyos/tests/smoke/cli/test_interfaces_bridge.py
test_add_multiple_ip_addresses (__main__.BridgeInterfaceTest.test_add_multiple_ip_addresses) ... ok
test_add_remove_bridge_member (__main__.BridgeInterfaceTest.test_add_remove_bridge_member) ... ok
test_add_single_ip_address (__main__.BridgeInterfaceTest.test_add_single_ip_address) ... ok
test_add_to_invalid_vrf (__main__.BridgeInterfaceTest.test_add_to_invalid_vrf) ... ok
test_bridge_delete_with_vxlan_heighbor_suppress (__main__.BridgeInterfaceTest.test_bridge_delete_with_vxlan_heighbor_suppress) ... ok
test_bridge_tunnel_vxlan_multicast (__main__.BridgeInterfaceTest.test_bridge_tunnel_vxlan_multicast) ... ok
test_bridge_vif_members (__main__.BridgeInterfaceTest.test_bridge_vif_members) ... ok
test_bridge_vif_s_vif_c_members (__main__.BridgeInterfaceTest.test_bridge_vif_s_vif_c_members) ... ok
test_bridge_vlan_filter (__main__.BridgeInterfaceTest.test_bridge_vlan_filter) ... ok
test_bridge_vlan_protocol (__main__.BridgeInterfaceTest.test_bridge_vlan_protocol) ... ok
test_dhcp_client_options (__main__.BridgeInterfaceTest.test_dhcp_client_options) ... ok
test_dhcp_disable_interface (__main__.BridgeInterfaceTest.test_dhcp_disable_interface) ... ok
test_dhcp_vrf (__main__.BridgeInterfaceTest.test_dhcp_vrf) ... ok
test_dhcpv6_client_options (__main__.BridgeInterfaceTest.test_dhcpv6_client_options) ... ok
test_dhcpv6_vrf (__main__.BridgeInterfaceTest.test_dhcpv6_vrf) ... ok
test_dhcpv6pd_auto_sla_id (__main__.BridgeInterfaceTest.test_dhcpv6pd_auto_sla_id) ... ok
test_dhcpv6pd_manual_sla_id (__main__.BridgeInterfaceTest.test_dhcpv6pd_manual_sla_id) ... ok
test_igmp_querier_snooping (__main__.BridgeInterfaceTest.test_igmp_querier_snooping) ... ok
test_interface_description (__main__.BridgeInterfaceTest.test_interface_description) ... ok
test_interface_disable (__main__.BridgeInterfaceTest.test_interface_disable) ... ok
test_interface_ip_options (__main__.BridgeInterfaceTest.test_interface_ip_options) ... ok
test_interface_ipv6_options (__main__.BridgeInterfaceTest.test_interface_ipv6_options) ... ok
test_interface_mtu (__main__.BridgeInterfaceTest.test_interface_mtu) ... ok
test_ipv6_link_local_address (__main__.BridgeInterfaceTest.test_ipv6_link_local_address) ... ok
test_isolated_interfaces (__main__.BridgeInterfaceTest.test_isolated_interfaces) ... ok
test_move_interface_between_vrf_instances (__main__.BridgeInterfaceTest.test_move_interface_between_vrf_instances) ... ok
test_mtu_1200_no_ipv6_interface (__main__.BridgeInterfaceTest.test_mtu_1200_no_ipv6_interface) ... ok
test_span_mirror (__main__.BridgeInterfaceTest.test_span_mirror) ... ok
test_vif_8021q_interfaces (__main__.BridgeInterfaceTest.test_vif_8021q_interfaces) ... ok
test_vif_8021q_lower_up_down (__main__.BridgeInterfaceTest.test_vif_8021q_lower_up_down) ... ok
test_vif_8021q_mtu_limits (__main__.BridgeInterfaceTest.test_vif_8021q_mtu_limits) ... ok
test_vif_8021q_qos_change (__main__.BridgeInterfaceTest.test_vif_8021q_qos_change) ... ok
test_vif_s_8021ad_vlan_interfaces (__main__.BridgeInterfaceTest.test_vif_s_8021ad_vlan_interfaces) ... skipped 'not supported'
test_vif_s_protocol_change (__main__.BridgeInterfaceTest.test_vif_s_protocol_change) ... skipped 'not supported'

----------------------------------------------------------------------
Ran 34 tests in 187.534s

OK (skipped=2)
```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/conf_mode/interfaces_bridge.py [hunk 1]
```diff
@@ -117,7 +119,20 @@ def get_config(config=None):
 
     return bridge
 
+def verify_vxlan_neighbor_suppress(bridge):
```
Comment (c-po) [inline src/conf_mode/interfaces_bridge.py]: Is there a reason for this to be a dedicated function? As this is not reused, it can happily live in verify()

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by c-po** [comment]
Please add a smoketest validating this szenario. It should fail before this commit and succeed with this fix. Other then that it looks good to me.

File: src/conf_mode/interfaces_bridge.py (hunk 1 above)
Comment (natali-rs1985) [reply src/conf_mode/interfaces_bridge.py]: Done

**Review by c-po — APPROVED** [review]

**Review by dmbaturin — APPROVED** [review]

**Comment by dmbaturin** [comment]
@mergifyio backport circinus

**Pull request merged**